FFMediaToolkit/Decoding/MediaFile.cs

```diff
@@ -17,38 +18,31 @@ private unsafe MediaFile(InputContainer container, MediaOptions options)
         {
             this.container = container;
 
-            if (container.Video != null)
+            var video = container.Decoders.Where(codec => codec.Info.Type == MediaType.Video);
+            var audio = container.Decoders.Where(codec => codec.Info.Type == MediaType.Audio);
+
+            if (video.Any())
             {
-                Video = new VideoStream(container.Video, options);
+                Video = video.Select(codec => new VideoStream(codec, options)).ToArray();
```

Comment:
I suppose you could just do this always and let Video be initialize as empty array, this would also fix some problems below where you acceess Video without checking for null

Reply:
Yeah, that's the idea.  My goal is to make the API as flexible as possible, part of that is handling edge cases in a way that is uniform and predictable.